tests/experimental/test_adjoint_jacobian_2.py

```diff
@@ -0,0 +1,669 @@
+    """Tests adjoint jacobian support"""
+
+    @pytest.fixture(params=[np.complex64, np.complex128])
+    def dev(self, request):
+        return LightningQubit2(c_dtype=request.param)
+
+    def test_not_expval(self, dev):
+        """Tests if a QuantumFunctionError is raised for a tape with measurements that are not
+        expectation values"""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.1, wires=0)
+            qml.var(qml.PauliZ(0))
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="Adjoint differentiation method does not"
+        ):
+            dev.preprocess(tape, AdjointConfig)
+
+    def test_CRot_unsupported(self, dev):
+        """Tests if a QuantumFunctionError is raised for an unsupported operation, i.e.,
+        multi-parameter operations that are not qml.Rot"""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.CRot(0.1, 0.2, 0.3, wires=[0, 1])
+            qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="The CRot operation is not supported using the"
+        ):
+            dev.preprocess(tape, AdjointConfig)
+
+    def test_Projector_unsupported(self, dev):
+        """Tests if a QuantumFunctionError is raised for a Projector observable"""
+        with qml.tape.QuantumTape() as tape:
+            qml.CRX(0.1, wires=[0, 1])
+            qml.expval(qml.Projector([0, 1], wires=[0, 1]))
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="differentiation method does not support the Projector"
+        ):
+            dev.preprocess(tape, AdjointConfig)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.CRX(0.1, wires=[0, 1])
+            qml.expval(qml.Projector([0], wires=[0]) @ qml.PauliZ(0))
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="differentiation method does not support the Projector"
+        ):
+            dev.preprocess(tape, AdjointConfig)
+
+    def test_state_return_type_unsupported(self, dev):
+        """Tests if a QuantumFunctionError is raised when the return type is State"""
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=[0])
+            qml.state()
+
+        tape.trainable_params = {0}
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="This method does not support statevector return type."
+        ):
+            dev.preprocess(tape, AdjointConfig)
+
+
+class TestAdjointJacobianComputeDerivatives:
+    """Tests for the adjoint_jacobian method"""
+
+    @pytest.fixture(params=[np.complex64, np.complex128])
+    def dev(self, request):
+        return LightningQubit2(c_dtype=request.param)
+
+    @staticmethod
+    def process_and_compute_derivatives(dev, tape):
+        batch, post_processing_fn = dev.preprocess(tape, AdjointConfig)
+        results = dev.compute_derivatives(batch, AdjointConfig)
+        return post_processing_fn(results)
```

Comment:
There is an intermittent error happening during these tests.
The error seems to indicate that the statevector provided to the tape is being mutated, instead of copied internally.
Since we rely on `create_initial_state`, the Core team is looking for a possible solution.